csharp/examples/OpenStudio.Tests/OpenStudio.Tests.csproj

```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>NET7</TargetFrameworks>
+    <TargetFrameworks>NET7;NET8;NET9;NET10</TargetFrameworks>
```

Comment:
This is my attempt at fixing the C# test step, just allow whatever is on the GHA runner.

Launch a run at https://github.com/NREL/OpenStudio/actions/runs/14217768843

Reply:
I thought this would allow testing with ANY version installed in that list, but it seems that it tries to do ALL of them so the workflow is probably going to fail.

```
  Determining projects to restore...
/usr/lib/dotnet/sdk/8.0.114/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(166,5): error NETSDK1045: The current .NET SDK does not support targeting .NET 9.0.  Either target .NET 8.0 or lower, or use a version of the .NET SDK that supports .NET 9.0. Download the .NET SDK from https://aka.ms/dotnet/download [/media/DataExt4/Software/Others/OS-build-csharp2/Testing/csharp/OpenStudio.Tests/OpenStudio.Tests.csproj::TargetFramework=NET9]
```

Reply:
MEH, just going to use NET8 and keep move on for now.

Launched at https://github.com/NREL/OpenStudio/actions/runs/14218506478

Reply:
Ok that run worked, including testing.